src/snowflake/snowpark/context.py

```diff
@@ -4,10 +4,22 @@
 #
 
 """Context module for Snowpark."""
+from typing import Callable, Optional
+
 import snowflake.snowpark
 
 _use_scoped_temp_objects = True
 
+# This is an internal-only function, used to interrupt registration of UDxF or SPROC objects with Snowflake.
+_internal_only_interrupt_registration: Optional[Callable] = None
+
+# This is an internal-only global flag, used to determine whether to execute code in a client's local sandbox or connect to a Snowflake account.
+_execute_in_local_sandbox: bool = False
+
+
```

Comment:
The two variables above will be set from an environment that invokes the python interpreter to read the user's files. For a lot of developers, this environment will be the one that Snowflake CLI is running in. In that case, we just want to fetch (validated and resolved) properties of the UDxF/SPROC decorator, and stop short of actually registering anything with Snowflake. Since this is an execution environment dependent behavior, the environment is responsible for setting the context here.